ocaml/runtime/memory.c

```diff
@@ -90,6 +90,14 @@ static struct page_table caml_page_table;
 
 int caml_page_table_lookup(void * addr)
 {
+#ifdef NO_NAKED_POINTERS
+  /* This case should only be hit if C stubs compiled without
+     NO_NAKED_POINTERS are linked into an executable using
+     "-runtime-variant nnp".  The return value here should cause the
+     macros in address_class.h to give the same results as when they
+     are compiled with NO_NAKED_POINTERS defined. */
+  return In_heap | In_young;
```

Comment:
I don't think this is enough. It gets the `Is_in_` macros right, but `Classify_addr` is also part of the public API.

I think the nnp version of this function should:
  - always return `In_heap`
  - never return `In_static_data`
  - return `In_young` according to `Is_young(addr)`
  - return `In_local` according to whether the pointer is in any of the local arenas

Reply:
Thanks, I've tried to implement this in the latest commit.